src/app/js/src/AddLayerDialog.jsx

```diff
@@ -1,28 +1,36 @@
 import React, { Component } from 'react';
-import { arrayOf, bool, func, object, string } from 'prop-types';
+import { arrayOf, bool, func, number, object, string } from 'prop-types';
 import { connect } from 'react-redux';
 
-import TileLayer from 'ol/layer/tile';
-import XYZ from 'ol/source/xyz';
-
 import Dialog from 'material-ui/Dialog';
 import FlatButton from 'material-ui/FlatButton';
 import TextField from 'material-ui/TextField';
 
 import {
     changeLayerName,
     changeLayerUrl,
-    changeTileJson,
     toggleAddLayerDialog,
+    toggleEditLayerDialog,
 } from './actions';
 
 class AddLayerDialog extends Component {
     constructor(props) {
         super(props);
+        // TODO: This should happen when this is opened, not when it is created
+        if (this.props.editMode && this.props.iToEdit !== -1) {
+            this.dispatch(changeLayerName({
+                name: this.props.layers[this.props.iToEdit].name,
+            }));
+            this.dispatch(changeLayerUrl({
+                url: this.props.layers[this.props.iToEdit].url,
```

Comment:
Consider replacing references to `i` with `id` or `layerId`. After reviewing this PR, it's clear what `i` means, but I think in the long run it would be more helpful if the variable name was longer.